Test install in appveyor

isuruf opened the pull request.

isuruf committed:
Test install in appveyor

isuruf committed:
Fix install

xoviat commented on .appveyor.yml:
```diff
@@ -48,7 +47,5 @@ build_script:
-  - cmd: copy lib\flang.lib %CONDA_INSTALL_LOCN%\Library\lib
-  - cmd: copy lib\ompstub.lib %CONDA_INSTALL_LOCN%\Library\lib
+  - cmd: cmake --build . --target install
+  - cmd: dir /s
```
Please make sure to find a command that is not so verbose, or upload a file to artifacts so that the log is readable.

isuruf replied:
This was just a test.

isuruf committed:
Fix Fortran compiler identification

isuruf commented on CMakeLists.txt:
```diff
@@ -25,6 +25,7 @@ SET(CMAKE_Fortran_ABI_COMPILED 0)
 SET(CMAKE_Fortran_COMPILER_SUPPORTS_F90 1)
 SET(CMAKE_Fortran_PREPROCESS_SOURCE
     "<CMAKE_Fortran_COMPILER> -cpp <DEFINES> <INCLUDES> <FLAGS> -E <SOURCE> -o <PREPROCESSED_SOURCE>")
+set(CMAKE_Fortran_MODDIR_FLAG "-module ")
```
@xoviat, https://gitlab.kitware.com/cmake/cmake/issues/17597 was because this was missing and the compiler identification is partial when flang is not fully built. When flang is fully built, this flag is set by CMake.

xoviat replied:
I also received the "old or corrupt module file" messages that you received here, caused by CMake not deleting the module file when it rebuilds the associated object. Any idea how to solve that?

isuruf committed:
Fix parallel build failures

This was because the static and shared libraries were both producing
.mod files at the same location and a parallel build was confusing
the compiler with the timestamps

isuruf commented:
I've removed https://github.com/isuruf/flang/pull/101 and fixed the old or corrupt failure. See https://github.com/isuruf/flang/pull/102/commits/23fd3a107b58d438a10151fe5ac7827753885cf6 for explanation.

isuruf committed:
reduce diff

isuruf committed:
Fix install failure

isuruf committed:
Add workaround for ninja

xoviat commented on runtime/flang/CMakeLists.txt:
```diff
@@ -498,21 +512,77 @@ else()
+)
+add_dependencies(flang_static
+  ieee_arithmetic
+)
```
These should depend of `flang1` and `flang2`

isuruf replied:
Thanks

isuruf commented:
@xoviat, instead of this way, we could have made `flang_shared` depend on `flang_static`.

ghost commented:
Tried that and got "corrupt or old module file"
